### PR TITLE
refactor(agent): gRPC communication with core

### PR DESCRIPTION
**Pull request by robot9706**
Implement *core* functionality to the gRPC communication to start phasing out the old REST interface.

### CODE REVIEW

File: agent/internal/grpc/grpc.go [hunk 1]
```diff
@@ -268,7 +276,12 @@ func executeVersionDeployRequest(
 		imageReq := mapper.MapDeployImage(req.Requests[i], appConfig)
 		dog.SetRequestID(imageReq.RequestID)
 
-		if err = deploy(ctx, dog, imageReq, &v1.VersionData{Version: req.VersionName, ReleaseNotes: req.ReleaseNotes}); err != nil {
+		var versionData *v1.VersionData = nil
```
Comment (nandor-magyar) [inline agent/internal/grpc/grpc.go:279]: default is nil for pointers, it is unnecessary to explicitly declare.

**Comment by nandor-magyar** [comment]
Please remove the unnecessary masrhal / unmarshal test functions.

File: protobuf/proto/agent.proto [hunk 2]
```diff
@@ -41,6 +41,8 @@ message AgentCommand {
   oneof command {
     VersionDeployRequest deploy = 1;
     ContainerStatusRequest containerStatus = 2;
+    ContainerDeleteRequest containerDelete = 3;
+    DeployRequestCore deployCore = 4;
```
Comment (nandor-magyar) [inline protobuf/proto/agent.proto:45]: Use of the word Legacy instead of Core? What do you think? Core might not say anything to the reader.

**Review by nandor-magyar — APPROVED** [review]

**Review by m8vago — CHANGES_REQUESTED** [review]
prename to prefix

**Review by m8vago — APPROVED** [review]

**Pull request merged**